Automattic/wp-calypso
# Help Center: calypso links open in same tab

Fixes #93429

## Proposed Changes

Extend Help Center article filter and action hooks to allow opening the Help Center to a specific article.

## Why are these changes being made?

Currently when the users clicks a link to Calypso from within the Help Center article it is opened in another tab. The better option would be to open the link in the same tab and persist the article. That is what we are doing here.

Since there was not an easy way to filter out only the calypso admin pages I opted to not include the site slug in the URL. This means they will need to select the site if its a URL like `https://wordpress.com/media/`, but it also means that it won't break URLs like `https://wordpress.com/learn`.

## Testing Instructions

1. Open the Help Center and click the article Getting Started
2. Scroll down to see the link for [WordPress.com plan](https://wordpress.com/plans/) and click that.
3. You should see it open in the same tab and the Help Center should open again to the article you were on.
4. Try a few other articles to see if other links work as well.
5. Regression test.
